mmdet3d/datasets/custom_3d.py

```diff
@@ -262,10 +265,31 @@ def evaluate(self,
             box_type_3d=self.box_type_3d,
             box_mode_3d=self.box_mode_3d)
         if show:
-            self.show(results, out_dir)
+            self.show(results, out_dir, pipeline=pipeline)
 
         return ret_dict
 
+    @staticmethod
+    def _extract_data(results, key):
+        """Extract and return the data corresponding to key in results dict.
+
+        Args:
+            results (dict): Result dict containing point clouds data.
+            key (str): Key of the desired data.
+
+        Returns:
+            np.ndarray | torch.Tensor: Data term.
+        """
+        # results[key] may be data or list[data]
+        # data may be wrapped inside DataContainer
+        data = results[key]
+        if isinstance(data, list) or isinstance(data, tuple):
+            data = data[0]
+        if isinstance(data, mmcv.parallel.DataContainer):
+            data = data._data
```

Comment:
Maybe add an else will be more safe and clear.

Reply:
I don't know, what should I do in else? I have taken ```data``` as ```result[key]```, even if the two ```if``` doesn't hit, we still have assigned value to ```data```. What do you think? Some advice please.